src/test/java/ro/msg/learning/shop/strategy/impl/SingleLocationStrategyTest.java

```diff
@@ -0,0 +1,130 @@
+package ro.msg.learning.shop.strategy.impl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ro.msg.learning.shop.entity.*;
+import ro.msg.learning.shop.entity.compositeId.StockId;
+import ro.msg.learning.shop.repository.OrdersRepository;
+import ro.msg.learning.shop.service.*;
+import ro.msg.learning.shop.util.OrderDetailsUtil;
+import ro.msg.learning.shop.util.ProductQuantityPairUtil;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class SingleLocationStrategyTest {
+
+    @Mock
+    private OrdersRepository ordersRepository;
+    @Mock
+    private LocationService locationService;
+    @Mock
+    private ProductService productService;
+    @Mock
+    private StockService stockService;
+    @Mock
+    private OrderDetailService orderDetailService;
+    @Mock
+    private CustomerService customerService;
+
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @Test
+    void handleOrderCreation() {
+        final Customer customer = new Customer();
+        customer.setOrdersList(new HashSet<>());
+        customer.setId(UUID.fromString("1484b5ef-a00d-4e4e-aebe-4ec17633574a"));
+
+        when(customerService.getCustomer(any())).thenReturn(customer);
+
+        final Set<Location> locations = new HashSet<>();
+
+        final Location location1 = new Location();
```

Comment:
Give the variables more meaningful names. Naming variables like var1, var2, ... can cause confusion and bugs in your code.